cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -229,16 +239,16 @@ func main() {
 
 	// Prepare http endpoint for metrics + leader election healthz
 	mux := http.NewServeMux()
-	if addr != "" {
-		metricsManager.RegisterToServer(mux, *metricsPath)
-		metricsManager.SetDriverName(provisionerName)
-		go func() {
-			klog.Infof("ServeMux listening at %q", addr)
-			err := http.ListenAndServe(addr, mux)
-			if err != nil {
-				klog.Fatalf("Failed to start HTTP server at specified address (%q) and metrics path (%q): %s", addr, *metricsPath, err)
-			}
-		}()
+	gatherers := prometheus.Gatherers{
+		// For workqueue and leader election metrics, set up via the anonymous imports of:
+		// https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/metrics/prometheus/workqueue/metrics.go
+		// https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/metrics/prometheus/clientgo/leaderelection/metrics.go
+		//
+		// Also to happens to include Go runtime and process metrics:
+		// https://github.com/kubernetes/kubernetes/blob/9780d88cb6a4b5b067256ecb4abf56892093ee87/staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go#L46-L49
```

Comment:
I once asked for component-base to include the whole set of runtime and process metrics here: https://github.com/kubernetes/kubernetes/issues/95152

But they decided not to include because of stability concern.

I guess that only leaves us the option to either manually add those metrics or use legacyregistry...? Dont really know how this will end up... It just feel messy for me overall for the metrics in k8s.

Reply:
> I once asked for component-base to include the whole set of runtime and process metrics here: kubernetes/kubernetes#95152
>
> But they decided not to include because of stability concern.

Hmm, isn't https://github.com/kubernetes/kubernetes/blob/9780d88cb6a4b5b067256ecb4abf56892093ee87/staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go#L46-L49 doing what you asked for?

Or do you want them to be added elsewhere?

I have not found documentation about the which role this metrics/legacyregistry package plays respectively is meant to play. I just figured out that in order to get access to workqueue metrics, I have to gather from that legacyregistry.

Reply:
Yes, this code you reference does what I asked for. But they are under this legacyregistry which implies this is not something we might want to use moving forward. So I was asking them to have something in the new registry.

I guess my main concern is that since we are using a lot legacyregistry. And I am afraid that for some later k8s release if the legacyregistry get deprecated and removed. We will have more work to do here. But this seems a bigger issue in the whole k8s